### PR TITLE
add new feature Poszer npc services

### PR DESCRIPTION
**Pull request by milestorme**
been using this on torn great upgrade to premium

### CODE REVIEW

**Comment by milestorme** [comment]
@wizzymore would be able to assist me with this travis error 
`/home/travis/build/azerothcore/azerothcore-wotlk/modules/mod-premium/src/Npc_Services.cpp:60:36: fatal error: unused variable 'ttr' [-Wunused-variable]
916                            uint32 ttr = (resetTime >= time(nullptr) ? resetTime - time(nullptr) : 0);
917                                   ^
9181 error generated.
919src/server/scripts/CMakeFiles/scripts.dir/build.make:803: recipe for target 'src/server/scripts/CMakeFiles/scripts.dir/__/__/__/modules/mod-premium/src/Npc_Services.cpp.o' failed
920make[2]: ***  `
We get the same error in mod-npc-services however I am not sure how to clean it , the instance reset works , just not sure how to redo this statement to fix ttr.

**Comment by milestorme** [comment]
❤️ thanks @wizzymore your on fire 🔥 great work , on everything. Legend .❤️

**Comment by milestorme** [comment]
Ah yes this error came up when I tryed fixing the first one too

**Comment by milestorme** [comment]
this is all mthe things that i ran into trying to fix it myself , which sucked since there 3 modules on azerothcore with this function

File: src/premium.cpp [hunk 1]
```diff
@@ -115,7 +124,10 @@ class premium_account : public ItemScript
             player->ADD_GOSSIP_ITEM(GOSSIP_ICON_TRAINER, GOSSIP_TEXT_TRAIN, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 8);
```
Comment (wizzymore) [inline src/premium.cpp]: Please use the following format for all gossip functions:
https://github.com/azerothcore/azerothcore-wotlk/blob/dc14094e28ea01448564c482dc9f63ca6b5ed2cf/src/server/game/AI/ScriptedAI/ScriptedGossip.h#L73

"AddGossipItemFor"
"SendGossipMenuFor"
And so on.

Comment (milestorme) [reply src/premium.cpp]: oof , i never planned to update the whole module like that , i was just adding a new feature , i dont even know how to do that ill have to look at another module , that will take me some time , im kind of a slow learner takes me a lot of time and repetition. maybe i can fully redo this module next year , this is how it is on azerothcore atm

Comment (wizzymore) [reply src/premium.cpp]: Yes indeed you are right, maybe i will fix them myself if i'll have some time

Thanks @milestorme

**Comment by milestorme** [comment]
I should probably test that the instance reset works ingame after these changes , did you test?

**Pull request closed without merging**